pilot/pkg/model/telemetry.go

```diff
@@ -410,32 +450,46 @@ func (t *Telemetries) telemetryFilters(proxy *Proxy, class networking.ListenerCl
 	return res
 }
 
-// mergeLogs returns the set of providers for the given logging configuration.
-// This currently is just the names of providers as there is no access logging configuration, but
-// in the future it will likely be extended
-func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig) (sets.Set, *tpb.AccessLogging_Filter) {
-	providers := sets.New()
+// mergeLogs returns the merged configuration for the given telemetry logging configuration.
+func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig) *LoggingSpec {
+	cfg := &LoggingSpec{
+		Client: &LoggingConfig{
```

Comment:
I do not see where you set LoggingConfig.Providers.  Is it useless?

Reply:
see L466, L502

Reply:
ProviderNames there

Reply:
IC, the providers are set in the caller, looks weird